src/preset.ts

```diff
@@ -564,6 +563,28 @@ function isInheritable(key: keyof ConfigurePreset | keyof BuildPreset | keyof Te
     return key !== 'name' && key !== 'hidden' && key !== 'inherits' && key !== 'description' && key !== 'displayName';
 }
 
+export function inheritsFromUserPreset(preset: ConfigurePreset | BuildPreset | TestPreset | PackagePreset | WorkflowPreset,
+    presetType: 'configurePresets' | 'buildPresets' | 'testPresets' | 'packagePresets' | 'workflowPresets', folderPath: string): boolean {
+
+    const originalUserPresetsFile: PresetsFile = getOriginalUserPresetsFile(folderPath) || { version: 8 };
+
+    const presetInherits = (presets: Preset[] | undefined, inherits: string | string[]) => presets?.find(p =>
+        Array.isArray(inherits)
+            ? inherits.some(inherit => inherit === p.name)
+            : inherits === p.name
+    );
+
+    if ((presetType === 'buildPresets' || presetType === 'testPresets' || presetType === 'packagePresets') &&
+        ((preset as ConfigurePresetInheritPreset).configurePreset &&
+            originalUserPresetsFile.configurePresets?.find(p => p.name === (preset as ConfigurePresetInheritPreset).configurePreset))) {
+        return true;
+    } else if (preset.inherits && presetInherits(originalUserPresetsFile[presetType], preset.inherits)) {
```

Comment:
From testing, I think we need a specific case for the "workflow preset". The workflow preset should not have the `inherit` property, it's actually an error that it extends Preset, because based on the docs here: https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#workflow-preset, it shouldn't. 

Therefore, we need to check the first step of the workflow, and the configure preset that it uses, when we do this

Reply:
Updated workflow preset to not extend Preset and removed the `inherit`, `environment`, `hidden`, `condition`, and `__inheritedPresetCondition` fields according to the cmake docs, so I had to take out some code from `expandWorkflowPreset` if you could double check that.